src/index.js

```diff
@@ -47,35 +47,45 @@ export default class SwupProgressPlugin extends Plugin {
 
 	stopShowingProgress = () => {
 		this.progressBar.setValue(1);
-		if( this.options.hideImmediately ) {
+		if (this.options.hideImmediately) {
 			this.hideProgressBar();
 		} else {
 			this.finishAnimationAndHideProgressBar();
 		}
-
-		if (this.showProgressBarTimeout != null) {
-			window.clearTimeout(this.showProgressBarTimeout);
-			delete this.showProgressBarTimeout;
-		}
 	};
 
 	showProgressBar = () => {
-		if (this.hideProgressBarTimeout != null) {
-			window.clearTimeout(this.hideProgressBarTimeout);
-			delete this.hideProgressBarTimeout;
-		}
+		this.cancelHideProgressBarTimeout();
 		this.progressBar.show();
 	};
 
 	showProgressBarAfterDelay = () => {
+		this.cancelShowProgressBarTimeout();
+		this.cancelHideProgressBarTimeout();
 		this.showProgressBarTimeout = window.setTimeout(this.showProgressBar, this.options.delay);
 	};
 
 	hideProgressBar = () => {
+		this.cancelShowProgressBarTimeout();
 		this.progressBar.hide();
 	};
 
 	finishAnimationAndHideProgressBar = () => {
+		this.cancelShowProgressBarTimeout();
 		this.hideProgressBarTimeout = window.setTimeout(this.hideProgressBar, this.options.transition);
 	};
+
+	cancelShowProgressBarTimeout = () => {
+		if (this.showProgressBarTimeout) {
```

Comment:
It's not necessary to check if a timeout is true or even set. See https://stackoverflow.com/questions/6442183/javascript-is-it-ok-to-call-cleartimeout-before-settimeout

Reply:
Even better!